js/coinflex.js

```diff
@@ -0,0 +1,1885 @@
+        //         "feeInstrumentId": "USD",
+        //         "orderId": "1002109741555",
+        //         "clientOrderId": "1651342025382"
+        //     }
+        //
+        // trades from order-object
+        //
+        //     {
+        //         "8375163007067827477": {
+        //             "matchQuantity": "334016",
+        //             "matchPrice": "0.00002089",
+        //             "timestamp": "1651410712318",
+        //             "orderMatchType": "TAKER"
+        //         }
+        //     }
+        //
+        const marketId = this.safeString (trade, 'marketCode');
+        market = this.safeMarket (marketId, market);
+        let id = undefined;
+        let timestamp = undefined;
+        let priceString = undefined;
+        let amountString = undefined;
+        let side = undefined;
+        let cost = undefined;
+        let fee = undefined;
+        let takerOrMakerRaw = undefined;
+        const keys = Object.keys (trade);
+        const length = keys.length;
+        if (length === 1) {
+            id = keys[0];
+            const tradeData = trade[id];
+            amountString = this.safeString (tradeData, 'matchQuantity');
+            priceString = this.safeString (tradeData, 'matchPrice');
+            cost = Precise.stringMul (amountString, priceString);
+            timestamp = this.safeInteger (tradeData, 'timestamp');
+            takerOrMakerRaw = this.safeString (trade, 'orderMatchType');
+        } else {
+            id = this.safeString (trade, 'matchId');
+            timestamp = this.safeInteger (trade, 'matchTimestamp');
+            priceString = this.safeString (trade, 'matchPrice');
+            amountString = this.safeString (trade, 'matchQuantity');
+            const sideRaw = this.safeString (trade, 'side');
+            side = this.parseOrderSide (sideRaw);
+            takerOrMakerRaw = this.safeString (trade, 'orderMatchType');
+            cost = this.safeNumber (trade, 'total');
+            const feeAmount = this.safeString (trade, 'fees');
+            if (feeAmount !== undefined) {
+                const feeCurrency = this.safeString (trade, 'feeInstrumentId');
+                fee = {
+                    'currency': this.safeCurrencyCode (feeCurrency, undefined),
+                    'cost': feeAmount,
+                };
+            }
+        }
+        const takerOrMaker = this.safeString ({ 'TAKER': 'taker', 'MAKER': 'maker' }, takerOrMakerRaw, 'taker');
+        return this.safeTrade ({
+            'id': id,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market['symbol'],
+            'order': this.safeString (trade, 'orderId'),
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': this.parseNumber (priceString),
+            'amount': this.parseNumber (amountString),
+            'cost': cost,
```

Comment:
`safeTrade` will initially convert these numbers to string so it is not worth parsing them as numbers here

Reply:
right! I didn't remembered its internals